### ISSUE TITLE
Sorry, I mean if `tmp_path` alone works? I check the definition of `AbstractConfigLoader` have a signature of `conf_source: str`, but I expect it should be `conf_source : str | Path`.

### ISSUE THREAD

**Issue by noklam** [kedro-org/kedro#2886]
Sorry, I mean if `tmp_path` alone works? I check the definition of `AbstractConfigLoader` have a signature of `conf_source: str`, but I expect it should be `conf_source : str | Path`.
![image](https://github.com/kedro-org/kedro/assets/18221871/140dbd86-cca9-4a5f-8dd9-3520b868c619)

A quick experiment suggests that it should work

![image](https://github.com/kedro-org/kedro/assets/18221871/2a53b31e-07a2-4b9e-a743-aead27de6f6f)

cc @merelcht 

_Originally posted by @noklam in https://github.com/kedro-org/kedro/pull/2869#discussion_r1280682742_

**Comment by Aftabby** [comment]
Hi, I'd like to work on this issue. would you please assign it to me? @astrojuanlu

**Comment by astrojuanlu** [comment]
Hi @Aftabby , thanks a lot! No need to assign it, just send the pull request

**Comment by astrojuanlu** [comment]
although there's already an open one addressing this issue

**Comment by Aftabby** [comment]
Yes, the last PR has some check errors with conflict merging. Can I still give it a try? @astrojuanlu

**Comment by merelcht** [comment]
Completed in https://github.com/kedro-org/kedro/pull/4637
